tests/integration_tests/saas/test_sendgrid_task.py

```diff
@@ -0,0 +1,138 @@
+    merged_graph = sendgrid_dataset_config.get_graph()
+    graph = DatasetGraph(merged_graph)
+
+    v = graph_task.run_access_request(
+        privacy_request,
+        policy,
+        graph,
+        [sendgrid_connection_config],
+        {"email": sendgrid_identity_email},
+    )
+
+    assert_rows_match(
+        v[f"{dataset_name}:contacts"],
+        min_size=1,
+        keys=[
+            "id",
+            "first_name",
+            "last_name",
+            "email",
+            "alternate_emails",
+            "address_line_1",
+            "address_line_2",
+            "city",
+            "state_province_region",
+            "country",
+            "postal_code",
+            "phone_number",
+            "whatsapp",
+            "list_ids",
+            "segment_ids",
+            "created_at",
+            "updated_at",
+        ],
+    )
+
+
+@pytest.mark.integration_saas
+@pytest.mark.integration_sendgrid
+def test_sendgrid_erasure_request_task(
+    db,
+    policy,
+    erasure_policy_string_rewrite,
+    sendgrid_secrets,
+    sendgrid_connection_config,
+    sendgrid_dataset_config,
+    sendgrid_erasure_identity_email,
+    sendgrid_erasure_data,
+) -> None:
+    """Full erasure request based on the sendgrid SaaS config"""
+    privacy_request = PrivacyRequest(
+        id=f"test_saas_erasure_request_task_{random.randint(0, 1000)}"
+    )
+    identity = PrivacyRequestIdentity(**{"email": sendgrid_erasure_identity_email})
+    privacy_request.cache_identity(identity)
+
+    dataset_name = sendgrid_connection_config.get_saas_config().fides_key
+    merged_graph = sendgrid_dataset_config.get_graph()
+    graph = DatasetGraph(merged_graph)
+
+    # access our erasure identity
+    v = graph_task.run_access_request(
+        privacy_request,
+        policy,
+        graph,
+        [sendgrid_connection_config],
+        {"email": sendgrid_erasure_identity_email},
+    )
+
+    # make sure erasure contact has expected fields
+    assert_rows_match(
+        v[f"{dataset_name}:contacts"],
+        min_size=1,
+        keys=[
+            "id",
+            "first_name",
+            "last_name",
+            "email",
+            "alternate_emails",
+            "address_line_1",
+            "address_line_2",
+            "city",
+            "state_province_region",
+            "country",
+            "postal_code",
+            "phone_number",
+            "whatsapp",
+            "list_ids",
+            "segment_ids",
+            "created_at",
+            "updated_at",
+        ],
+    )
+    config.execution.MASKING_STRICT = False  # Allow delete
+    erasure = graph_task.run_erasure(
+        privacy_request,
+        erasure_policy_string_rewrite,
+        graph,
+        [sendgrid_connection_config],
+        {"email": sendgrid_erasure_identity_email},
+        get_cached_data_for_erasures(privacy_request.id),
+    )
+    assert erasure == {"sendgrid_connector_example:contacts": 1}
```

Comment:
do we know how long deletion takes? if it's a matter of seconds (rather than minutes/hours), then it's probably worth us invoking another `poll_for_existence` here to confirm that the the contact does _not_ exist anymore, i.e. confirming that our deletion was really successful in the external system.

you'd be able to invoke it similarly to how it's being invoked in `sendgrid_fixtures.py` now, but just passing `False` for the `existence_desired` argument and updating the `error_message` appropriately. (you'd also probably want to update the `_contact_exists` method in that module to no longer have a leading `_` since it'd not solely be for internal module use.